src/createAnimatedComponent/inlinePropUpdater.ts

```diff
@@ -0,0 +1,190 @@
+  _inlineProps: StyleProps = {};
+
+  public static hasInlineStyles(style: StyleProps): boolean {
+    if (!style) {
+      return false;
+    }
+    return Object.keys(style).some((key) => {
+      const styleValue = style[key];
+      return (
+        isSharedValue(styleValue) ||
+        (key === 'transform' && isInlineStyleTransform(styleValue))
+      );
+    });
+  }
+
+  public attachInlineProps(
+    animatedComponent: React.Component<unknown, unknown>,
+    viewInfo: any // viewTag, viewName, shadowNodeWrapper, viewConfig
+  ) {
+    const newInlineProps: Record<string, any> = extractSharedValuesMapFromProps(
+      animatedComponent.props
+    );
+    const hasChanged = inlinePropsHasChanged(newInlineProps, this._inlineProps);
+
+    if (hasChanged) {
+      if (!this._inlinePropsViewDescriptors) {
+        this._inlinePropsViewDescriptors = makeViewDescriptorsSet();
+
+        const { viewTag, viewName, shadowNodeWrapper, viewConfig } = viewInfo;
+
+        if (Object.keys(newInlineProps).length && viewConfig) {
+          adaptViewConfig(viewConfig);
+        }
+
+        this._inlinePropsViewDescriptors.add({
+          // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+          tag: viewTag as number,
+          // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+          name: viewName!,
+          // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+          shadowNodeWrapper: shadowNodeWrapper!,
+        });
+      }
+      const sharableViewDescriptors =
+        this._inlinePropsViewDescriptors.sharableViewDescriptors;
+
+      const maybeViewRef = NativeReanimatedModule.native
+        ? undefined
+        : ({ items: new Set([this]) } as ViewRefSet<any>); // see makeViewsRefSet
+
+      const updaterFunction = () => {
+        'worklet';
+        const update = getInlinePropsUpdate(newInlineProps);
+        updateProps(sharableViewDescriptors, update, maybeViewRef);
+      };
+      this._inlineProps = newInlineProps;
+      if (this._inlinePropsMapperId) {
+        stopMapper(this._inlinePropsMapperId);
+      }
+      this._inlinePropsMapperId = null;
+      if (Object.keys(newInlineProps).length) {
+        this._inlinePropsMapperId = startMapper(
+          updaterFunction,
+          Object.values(newInlineProps)
+        );
+      }
+    }
+  }
+
+  public detachInlineProps() {
+    if (this._inlinePropsMapperId) {
+      stopMapper(this._inlinePropsMapperId);
+    }
+  }
+
+  public getInlineStyle(
+    style: Record<string, unknown>,
+    isFirstRender: boolean
+  ) {
+    if (isFirstRender) {
+      return getInlinePropsUpdate(style);
+    }
+    const newStyle: StyleProps = {};
+    for (const [key, styleValue] of Object.entries(style)) {
+      if (
+        !isSharedValue(styleValue) &&
+        !(key === 'transform' && isInlineStyleTransform(styleValue))
+      ) {
+        newStyle[key] = styleValue;
+      }
+    }
+    return newStyle;
+  }
```

Comment:
Why is `hasInlineStyles` a static method and this is non-static?

Reply:
Good point, I'll fix it

Reply:
I think that actually only `attach` and `detach` should be the members of this class - the rest of the functions present in this file should either all be static methods or outside of this class and I think the former is better.